sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/rust/ToRustInnerVisitor.java

```diff
@@ -458,9 +458,18 @@ public VisitDecision preorder(DBSPDecimalLiteral literal) {
         if (type.mayBeNull)
             this.builder.append("Some(");
         String value = Objects.requireNonNull(literal.value).toPlainString();
-        this.builder.append("Decimal::from_str(\"")
+
+        int scale = (type.precision == DBSPTypeDecimal.MAX_PRECISION && type.scale == 0) ? DBSPTypeDecimal.MAX_SCALE : type.scale;
```

Comment:
why these conditionals? Why not just type.scale?

Reply:
This was to try and match the Postgres behavior and avoid something like `2.1::DECIMAL` giving `2`. 
But now that we decided to match the SQL specification and not match Postgres, this can be removed.